common/ffsparser.cpp

```diff
@@ -164,7 +160,7 @@ USTATUS FfsParser::parseGenericImage(const UByteArray & buffer, const UINT32 loc
 {
     // Parse as generic UEFI image
     UString name("UEFI image");
-    UString info = usprintf("Full size: %Xh (%u)", buffer.size(), buffer.size());
+    UString info = usprintf("Full size: %" PRIXQ "h (%" PRIuQ ")", buffer.size(), buffer.size());
```

Comment:
I know this things are better, but they are also ugly as all hell.

Reply:
Alternative is to typecast to UINT32 as was done in 34c8ad8dcc5f4bd76509123626d0f00511d3eb45.
Or use a different function for printing output, such as the {fmt} library https://github.com/fmtlib/fmt

Reply:
I generally recommend type casts, because we have predictable sizes of the values. Perhaps not to `UINT32` but to `unsigned`, because `UINT32` can be `unsigned long` or `unsigned int` depending on the compiler (e.g. gcc vs clang on i386).

Reply:
I think I prefer the format macros. I understand all sizes will probably be less than 32 bit but I don't like typecasting to a smaller size if there's an alternative.

If you decided to go with type cast, I would probably use the separate header uinttypes.h to define the type that you want to type cast to. Something like:
`typedef unsigned QINT`

I think `unsigned` is the correct type to use because the lack of a size qualifier is similar to the lack of size qualifier in the printf formats `%X` and `%u`.

If you want to make sure `unsigned` is always greater than 16 bits, then you can add a statement that would cause a compiler error.
`char testuinttypes[sizeof(unsigned) >= 4 ? 1 : -1];`